pkg/schema/annotations.go

```diff
@@ -170,26 +174,36 @@ func collectAnnotations(item yamlmeta.Node) ([]Annotation, error) {
 	return anns, nil
 }
 
-func processOptionalAnnotation(node yamlmeta.Node, optionalAnnotation template.AnnotationName) (Annotation, error) {
-	nodeAnnotations := template.NewAnnotations(node)
-
-	if nodeAnnotations.Has(optionalAnnotation) {
-		ann, _ := nodeAnnotations[optionalAnnotation]
+func collectValueAnnotations(item yamlmeta.Node, t yamlmeta.Type) ([]Annotation, error) {
+	var anns []Annotation
 
-		wrappedValueType, err := inferTypeFromValue(node.GetValues()[0], node.GetPosition())
+	for _, annotation := range []template.AnnotationName{AnnotationNullable, AnnotationDefault} {
+		ann, err := processOptionalAnnotation(item, annotation, t)
 		if err != nil {
 			return nil, err
 		}
+		if ann != nil {
+			anns = append(anns, ann)
+		}
+	}
+	return anns, nil
+}
+
+func processOptionalAnnotation(node yamlmeta.Node, optionalAnnotation template.AnnotationName, t yamlmeta.Type) (Annotation, error) {
+	nodeAnnotations := template.NewAnnotations(node)
+
+	if nodeAnnotations.Has(optionalAnnotation) {
+		ann, _ := nodeAnnotations[optionalAnnotation]
```

Comment:
```suggestion
		ann := nodeAnnotations[optionalAnnotation]
```
nit: we're ignoring the `ok` return param, anyway; we can just not include it. 👍🏻  (it was there before, but why not clean it up)